### ISSUE TITLE
Add persistent option for Default Printer

### ISSUE THREAD

**Issue by cleidigh** [cleidigh/printing-tools-ng#188]
@Thefaroah
@kiki-ja 
@RRoenn 
@tidebre67 
@JF1313 
@DenB10

Added in b4 allowing PTNG to make the Default printer in PTNG to be sticky and set for each print regardless if changed in preview.
@cleidigh

**Comment by Thefaroah** [comment]
Few issues:
1. Started with the following settings:
![image](https://user-images.githubusercontent.com/46969365/230790319-15714e49-43db-4fb2-b254-40792bddf866.png)
2. Then printed to TB, changing the printer:
![image](https://user-images.githubusercontent.com/46969365/230790357-aeb00f8b-6a1c-48f6-99e5-f618e6a1f79d.png)
3. Then went back into PTNG settings:
![image](https://user-images.githubusercontent.com/46969365/230790383-6b2d9dbc-8391-485b-9627-d76f07b2d815.png)
Also, the print without dialog window option  acts as if it were checked (doesn't ask for the "Click OK.....)
![image](https://user-images.githubusercontent.com/46969365/230790515-c1cc0e50-38b0-4514-978a-1bbce84ed85a.png)
If I restart from Step 1 above, the "Click OK..... reappears.
Boy, I hope I explained things clear enough.

**Comment by cleidigh** [comment]
@Thefaroah 
What if you try again, I wonder if I have a first time issue?
The option only appears for pdf output.
@cleidigh

**Comment by Thefaroah** [comment]
You're right!! I should have restarted TB before I wrote you.
All appears as expected, especially since you explained "The option only appears for pdf output."
Tried every combination of options and couldn't break it. I think it works fine.

**Comment by JF1313** [comment]
@cleidigh fantastic change! Thank you. Where can I find the link to download?

**Comment by cleidigh** [comment]
@JF1313 
I always have the latest beta at the top of the release status 
- #189 
- @cleidigh

**Comment by Thefaroah** [comment]
I'll never know now. I've started/restarted TB and the system a couple times for other reasons.
PTNG has behaved... I like it.
Anything else needing my help? I'm getting bored. :)

**Comment by cleidigh** [comment]
@Thefaroah 
Well not yet, once I get something done to try and fix this you can help verify. We will have to start from a new profile. There is always more especially when we update for 115, which I have started.
@cleidigh

**Comment by Thefaroah** [comment]
Let me know.
BTW... I'm not too brave venturing beyond TB current releases.

**Comment by cleidigh** [comment]
@Thefaroah 
I cannot seem to reproduce this error on upgrading.
@kiki-ja 
@JF1313 
@RRoenn 
@tidebre67 
If you guys upgrade and try the persistent option, note if it appears not to work and requires a Thunderbird restart.
My tests have not required it, but I need others to verify.
@cleidigh

**Comment by JF1313** [comment]
@cleidigh thanks for the lihk. I've downloaded the beta and will update you i anything out of the ordinary happens.

**Comment by JF1313** [comment]
@cleidigh thanks for the lihk. I've downloaded the beta and have selected the persistent option. Should anything out of the ordinary appear I'll let you know. Thanks for all your help!

**Comment by cleidigh** [comment]
@JF1313 
Thx
@cleidigh

**Comment by tidebre67** [comment]
> @Thefaroah I cannot seem to reproduce this error on upgrading. @kiki-ja @JF1313 @RRoenn @tidebre67 If you guys upgrade and try the persistent option, note if it appears not to work and requires a Thunderbird restart. My tests have not required it, but I need others to verify. @cleidigh

@cleidigh 
as far as i can see it works correct.

**Comment by cleidigh** [comment]
@Thefaroah 
Ok I cannot seem to reproduce this.
As I understand you this state is now "persistent", yes pun intended.
Can you check the preferences in the config editor?
I am looking for the following entries : 
print_printer
extensions.printingtoolsng.print_printer
extensions.printingtoolsng.printer.persistent

I may have to add debug...
@cleidigh

**Comment by Thefaroah** [comment]
Here they are, along with current setting:
![image](https://user-images.githubusercontent.com/46969365/231302950-7d248ceb-015e-4c95-b876-0e88b876ee44.png)

print_printer	Mozilla Save to PDF
extensions.printingtoolsng.print_printer	Mozilla Save to PDF
extensions.printingtoolsng.printer.persistent	true

**Comment by cleidigh** [comment]
@Thefaroah 
All as expected.
I will have to add debug 
@cleidigh

**Comment by Thefaroah** [comment]
Perhaps repeating my steps:
Persistent printer set to:
![image](https://user-images.githubusercontent.com/46969365/231310829-c8438088-4429-4cf6-9c43-99991598c70f.png)
Enter TB print preview and set printer to Fineprint and print:
![image](https://user-images.githubusercontent.com/46969365/231310924-72408619-1c37-4795-aebb-15a3eda13803.png)
Back to PTNG settings:
![image](https://user-images.githubusercontent.com/46969365/231311004-84e82f5c-4ae1-4116-b3c0-da342b652e0f.png)
By any chance is there an issue flipping between PDF and non-PDF printers?

**Comment by cleidigh** [comment]
@Thefaroah 
Should not have anything to do with the printer. I will scour when I add debug.
Basically I maintain the parallel printer setting only controlled by ptng. Then every time a print is started, check persistent flag and if set change printer to ptng printer.
It's fairly basic, just can't see why it would break.
We will figure it out, just need some of your help to debug.
Thx
@cleidigh

**Comment by Thefaroah** [comment]
When you're ready. May need some help obtaining debug info.

**Comment by cleidigh** [comment]
@Thefaroah 
Ok I have added some debug.
So load b5 and then do the following : 
- open options and set debug options to:
trace1 printsettings 
- Open console and clear 
- follow your prior sequence 
- go to console, select all with right click select all, copy
- save as txt and post here 

We should see printer forced to Default 
@cleidigh

**Comment by Thefaroah** [comment]
Where is console?

**Comment by cleidigh** [comment]
@Thefaroah 
Sorry, made assumptions 
Use the shortcut 
Shift-Control-J
If you close tb you have to close the console separately 
trash can clears
@cleidigh

**Comment by Thefaroah** [comment]
TypeError: browser is null
[BrowserElementParent.jsm:24:21](resource://gre/actors/BrowserElementParent.jsm)
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[{1ac19d05-b4b1-4215-85da-f14ecaf0eb02}](moz-nullprincipal:{1ac19d05-b4b1-4215-85da-f14ecaf0eb02})
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[blank](about:blank)
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[Inbox](mailbox:///C:/Users/Mike/AppData/Roaming/Thunderbird/Profiles/ocbyx29e.default/Mail/TheFaroah%20(G)/Inbox?number=126)
Cookie “” has been rejected as third-party. [ALGLEFKWBNT2FI2CP5FCY5TXA366HA5CNFSM6AAAAAAWHCCAWKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSZYDNEO.gif](https://github.com/notifications/beacon/ALGLEFKWBNT2FI2CP5FCY5TXA366HA5CNFSM6AAAAAAWHCCAWKWGG33NNVSW45C7OR4XAZNMJFZXG5LFINXW23LFNZ2KUY3PNVWWK3TUL5UWJTSZYDNEO.gif)
Request to access cookie or storage on “<URL>” was blocked because we are blocking all third-party storage access requests and content blocking is enabled. 2
Cookie “logged_in=no; domain=github.com; path=/; expires=Fri, 12 Apr 2024 19:52:52 GMT; secure; HttpOnly; SameSite=Lax
_gh_sess=pRh8aBXs%2FpMetFKQT8k4pslQcA5sI4M6Sg6urh1EVwVF66bapXGTGR8FqT35t8H7JX1WLWNLqSRGPomKIG7UIQYYKj8Rt6P9qvlq8TxQH2xFmo1p2dj9X96sMhx%2Fj7qsje%2BjvMR5RrJBerEJRs%2BYX1kFbyI1pVLsZI%2BNNQQ1pjEQlcbrc3Vzj8RGeNtrIH%2F19fiRDGH0U%2FfVfrrsGEy3mWRNscPn5yrf0WvxJRXO9MKxuceUU0Yt9o2O4FoCaBo9onUpajHuLQcBxDagzgikwEt9pC0vKqM%2F132%2FEDu%2BfjMFuTYpIEOqSxcinqprnsi7fPxYZRFKg6l8b8wJBYwffMrannJEek6oJe%2FpN344ZDgIcNmyKbWG44TzEtOW%2FKX1lOn7IYHHqgVSgR2nF9dthYFmI2SzK18y3Mn5hdwGxXZDCvJz0gLbcvLB52WcFh8rWq8tX60HuharjG3u2fCmX9FyQgiDu0GG7SQno3fPkBZEn45GxB7enfC41aIEMw6ddBaHF7CxqkgkBrfD9PMXKPHOP5LPXVhOr%2BWEowlsNHlhkC1z4A0%2Feg%2BFVqqVmR4ApcpjzspHVmbjsaMs%2BTNzYw3b3S49OCRCIJ9%2FkB9lMf%2FavT7L70Fqhnsil2h9Hy3BiVmjgMct9hLD9CsLBNlh3%2Fc3boRxZ6eOdcUkYfdppNawTA2qmCz6RP92ojJZvUz5b9mmTIPDg%2BS3%2FzN6tAiP3RHImjKiJ1ktQpMIKP%2Fb6KBWSq%2BIPSeORCgFzTv4Fj5PPbqZZ4xvMGjBZ8EYMRieqFC6KK2ujzyF1My7QrFBFyFZOacnAX0%2FBjfY2dQYhcI708rV3B0L4r6mMK6QN2myt%2BZhQj0Tf%2BG7tGHwpwE258KDaFPGA%2Bn1KxdDV0tZdTHctkF3By2i6Q2DVMMuaZ62eFKcvlehNPBiTWqyh2jcRKbyijVgmXCMk3Rdsf%2FGyvh869jR88rUh7E7adCSeRd%2FUfDWxppX4bbPAJqkUQxrf3Slpw3AqOs2dtnyY0o%2F4klwrPQrXrzBnHCwSUzIrojrPjE%2FHOEO%2BmPEF%2Frm7xeMqzN642odgvVh0REvw4gds%2FC6JHyPjBnCmZgEvMMQvYaCkzPlrPeNlAGWCU%2BEZ0TrCm4m85Gu3pULoaAFYPV1jBTVOh6mK8BSX7y%2FDvvoYM5M35KNNrh5seC9LVNSPabkbJnTL7E4sGO3wFsBKdKLDA7TKZnq8Oipn0nnqazlEeGWbE7LUgIpAOGQgvikwQd375CFdL83hjJgkRgRW0iPJe%2FIEPs1o41ZvyDd0pRq3XAsuMWOe4d13lxPxioIMcKxDfU6hvkKcH%2F2hHl3E4e83WeIjqsagsxTXnrzr9fipmNd3HehBSdFoPGu6aziWzC%2Bn2eTpQkPrXVzQ6aYpkXu0EjFVBCN5FnH9gW3SsV0Kpil3q2P3LaqbllK64y6Yg2r%2FbXhd2sKDMw8dYStQy8%2Fhs7SEKVETZUstOK47L9vjtAdlkqJkE1sz9ermY5wFngUP1vBn1LueGJbpMJtoTjsfULrGi2NXRcHJDZkwAQ7nwyyykPNDMUZ1Nug0B%2BrjQKSWsHwS2ITgYOtADBvv7%2Fk17hR7n%2Fdxu5E9u7k8xBWnltXIbwssBGnFhQUdRsrDm2fSEF1ijGrcKUK9P%2BSaDrunf5DrIgkVltM2T%2FuHlVmeANYgkdSkeaL1w0j9YHnHXohQTpPtAHgfzgouYcIGMQ%2FCj3vK8oPa1QVUKQEh57QlVdTfPyE3mY5XUZc5XKTgiBg6fjGjKU3zQOnQp8EXkPpPJ73e0uUDr8E1MHG--S8aDNEJeO%2BgxcxX9--ALQO7hMkD1oxnHYItwjkKA%3D%3D; path=/; secure; HttpOnly; SameSite=Lax” has been rejected as third-party.

**Comment by cleidigh** [comment]
ok clear that then follow your prior steps printing 
we should see ptng output 
@cleidigh

**Comment by Thefaroah** [comment]
TypeError: browser is null BrowserElementParent.jsm:24:21

**Comment by cleidigh** [comment]
I am confused, if you tried a print there should be lots of debug output.
You have the console open, do the printing then return to console and copy?

**Comment by cleidigh** [comment]
in the top right of the console are all the message types enabled?

**Comment by cleidigh** [comment]
here is a screenshot of the end of one print 
<img width="960" alt="2023-04-12-17-08-01 " src="https://user-images.githubusercontent.com/25272315/231585721-89cb7902-b06b-4123-8730-cbbfc0fb973e.png">

**Comment by Thefaroah** [comment]
Only some of the types were hi-lited. I checked the rest and got the following:
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘widthx’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘vertical-alignment’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘vertical-align’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Element.releaseCapture() is deprecated. Use Element.releasePointerCapture() instead. For more help https://developer.mozilla.org/docs/Web/API/Element/releasePointerCapture [menupopup.js:169:13](chrome://global/content/elements/menupopup.js)
XHRPOSThttps://www.googleapis.com/oauth2/v3/token
[HTTP/3 200 OK 216ms]

PTNG: cmd_printng start: options :  null [printingtoolsng-pengine.js:578:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: current window:  
Object { id: 3, focused: true, top: 2, left: 2396, width: 1444, height: 984, incognito: false, type: "normal", state: "normal", alwaysOnTop: false, … }
[printingtoolsng-pengine.js:596:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: current tab:  
Object { id: 1, index: 0, windowId: 3, highlighted: true, active: true, status: "complete", width: 1088, height: 578, url: "mailbox:///C:/Users/Mike/AppData/Roaming/Thunderbird/Profiles/ocbyx29e.default/Mail/TheFaroah%20(G)/Inbox?number=131", title: "Inbox - Gmail", … }
[printingtoolsng-pengine.js:597:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: tab type :  mail [printingtoolsng-pengine.js:598:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: tab url:  mailbox:///C:/Users/Mike/AppData/Roaming/Thunderbird/Profiles/ocbyx29e.default/Mail/TheFaroah%20(G)/Inbox?number=131 [printingtoolsng-pengine.js:599:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: mailType:  true [printingtoolsng-pengine.js:600:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: selectedMessageUris 
Array [ "mailbox-message://thefaroah%40gmail.com@pop.gmail.com/Inbox#131" ]
[printingtoolsng-pengine.js:63:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
forcePrinterToPTNGPrinter: start [printerSettings.js:545:13](chrome://printingtoolsng/content/printerSettings.js)
current system printer (bef force): Mozilla Save to PDF [printerSettings.js:546:13](chrome://printingtoolsng/content/printerSettings.js)
force printer to : Mozilla Save to PDF [printerSettings.js:550:15](chrome://printingtoolsng/content/printerSettings.js)
current system printer : Mozilla Save to PDF [printerSettings.js:551:15](chrome://printingtoolsng/content/printerSettings.js)
Mozilla Save to PDF [printerSettings.js:360:13](chrome://printingtoolsng/content/printerSettings.js)
Current printer before print:  Mozilla Save to PDF [printingtoolsng-pengine.js:251:13](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: Done [printingtoolsng-pengine.js:622:11](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
Message pane focused  [printingtoolsng-pengine.js:182:14](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
TypeError: browser is null
[BrowserElementParent.jsm:24:21](resource://gre/actors/BrowserElementParent.jsm)
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘widthx’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘vertical-alignment’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘vertical-align’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)

​

**Comment by cleidigh** [comment]
ok now we have debug output 
everything looks good, can you capture when it does not set the printer correctly?

**Comment by Thefaroah** [comment]
Set the "persistent" to Save to PDF.
Cleared the log. (Started logging)
Clicked the Print NG icon, cancelled the ensuing pop-up.
Changed the Destination printer to FinePrint.
Clicked Print.
Went back to PTNG options.
Persistent printer became FinePrint, not Print to PDF (as, I assume, it should be)
Copied the console entries.
Here's the console log;
PTNG: cmd_printng start: options :  null [printingtoolsng-pengine.js:578:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: current window:  
Object { id: 3, focused: true, top: 2, left: 2396, width: 1444, height: 984, incognito: false, type: "normal", state: "normal", alwaysOnTop: false, … }
[printingtoolsng-pengine.js:596:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: current tab:  
Object { id: 1, index: 0, windowId: 3, highlighted: true, active: true, status: "complete", width: 1088, height: 578, url: "mailbox:///C:/Users/Mike/AppData/Roaming/Thunderbird/Profiles/ocbyx29e.default/Mail/TheFaroah%20(G)/Inbox?number=132", title: "Inbox - Gmail", … }
[printingtoolsng-pengine.js:597:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: tab type :  mail [printingtoolsng-pengine.js:598:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: tab url:  mailbox:///C:/Users/Mike/AppData/Roaming/Thunderbird/Profiles/ocbyx29e.default/Mail/TheFaroah%20(G)/Inbox?number=132 [printingtoolsng-pengine.js:599:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: mailType:  true [printingtoolsng-pengine.js:600:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: selectedMessageUris 
Array [ "mailbox-message://thefaroah%40gmail.com@pop.gmail.com/Inbox#132" ]
[printingtoolsng-pengine.js:63:12](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
forcePrinterToPTNGPrinter: start [printerSettings.js:545:13](chrome://printingtoolsng/content/printerSettings.js)
current system printer (bef force): Mozilla Save to PDF [printerSettings.js:546:13](chrome://printingtoolsng/content/printerSettings.js)
force printer to : Mozilla Save to PDF [printerSettings.js:550:15](chrome://printingtoolsng/content/printerSettings.js)
current system printer : Mozilla Save to PDF [printerSettings.js:551:15](chrome://printingtoolsng/content/printerSettings.js)
Mozilla Save to PDF [printerSettings.js:360:13](chrome://printingtoolsng/content/printerSettings.js)
Current printer before print:  Mozilla Save to PDF [printingtoolsng-pengine.js:251:13](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
PTNG: Done [printingtoolsng-pengine.js:622:11](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
Message pane focused  [printingtoolsng-pengine.js:182:14](chrome://printingtoolsng/content/printingtoolsng-pengine.js)
TypeError: browser is null
[BrowserElementParent.jsm:24:21](resource://gre/actors/BrowserElementParent.jsm)
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘widthx’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Unknown property ‘vertical-alignment’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘background-color’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Error in parsing value for ‘vertical-align’.  Declaration dropped. [ptng-options.xhtml](chrome://printingtoolsng/content/ptng-options.xhtml)
Element.releaseCapture() is deprecated. Use Element.releasePointerCapture() instead. For more help https://developer.mozilla.org/docs/Web/API/Element/releasePointerCapture [menupopup.js:169:13](chrome://global/content/elements/menupopup.js)

**Comment by cleidigh** [comment]
From the above it is printing to save as pdf, that is what was set, what am I missing?

**Comment by Thefaroah** [comment]
When the TB print preview appears, I changed the Print to PDF to FinePrint, then print:
![image](https://user-images.githubusercontent.com/46969365/231602293-b8f750f3-62ac-463a-83ff-fd297417f11c.png)
Then going back to PTNG options, my persistent printer (save to PDF) has been changed to FinePrint:
![image](https://user-images.githubusercontent.com/46969365/231602490-5cc356d2-1110-4d42-8444-597b9e324a80.png)

**Comment by cleidigh** [comment]
alright my bad.
The printing is correct, but ptng shows the wrong printer? Or would the next print go to Fineprint?

**Comment by Thefaroah** [comment]
The next print would go to FinePrint

**Comment by cleidigh** [comment]
ok i think I reproduced it!
I have to go offline for a while, but I have it to work on.
Thanks, sorry it was tricky.
@cleidigh

**Comment by Thefaroah** [comment]
I may have stumbled across something:
Set persistent to "save as PDF"
PTNG icon, TB Print, change to FinePrint, print.
DO NOT go into PTNG options.
Click the PTNG icon and all appears OK (save as PDF shows).

Now, I repeated all of the above, except I DID go into PTNG options, and persistent was incorrect (FinePrint)

**Comment by cleidigh** [comment]
yeah I think the ptng options is not forcing the reset so that is a different path.

**Comment by Thefaroah** [comment]
was about to suggest (in layman's terms), the options setting was looking in the wrong spot for the persistent printer name.

**Comment by cleidigh** [comment]
@Thefaroah 
ok I still don't get why I missed this...
b6 fills the hole.
@cleidigh

**Comment by Thefaroah** [comment]
Tried to break it... no success.
I think you've got it. Congratulations!!!

**Comment by cleidigh** [comment]
well you just have to try harder :-)...

**Comment by Thefaroah** [comment]
Just did. Even with a TB restart.
Again, no success in breaking it.
Now, on to your next project? Keep me in mind.

**Comment by cleidigh** [comment]
After this it's 115 update for IETNG then back to ptng

**Comment by Thefaroah** [comment]
OK. Good luck. And if you need me, you know how to find me.

**Comment by cleidigh** [comment]
sounds good, I have you on the credits list, you are officially in the merry band of ptng crew members!

**Comment by Thefaroah** [comment]
Cool!!!
Just curious. What do you have in mind for PTNG in the future?

**Comment by cleidigh** [comment]
well basically it's feature freeze for a while then I really have to see what users want. Things with attachments have been requested.

**Comment by Thefaroah** [comment]
I'll try to think of a thing or two. Don't want you to get bored.
Take a break. You earned it.

**Comment by cleidigh** [comment]
thx

**Comment by JF1313** [comment]
@cledigh As always, thank you for staying on top of the details and fixing everything!!
